src/hotspot/cpu/x86/x86.ad

```diff
@@ -8633,19 +8627,17 @@ instruct vmuladdaddS2I_reg(vec dst, vec src1, vec src2) %{
 %}
 
 // --------------------------------- PopCount --------------------------------------
-
 instruct vpopcount_integral_reg_evex(vec dst, vec src) %{
-  predicate((((Matcher::vector_element_basic_type(n->in(1)) == T_INT ||
-               Matcher::vector_element_basic_type(n->in(1)) == T_LONG)) &&
-             VM_Version::supports_avx512_vpopcntdq()) ||
-            (is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
-             VM_Version::supports_avx512_bitalg()));
+  predicate(UsePopCountInstruction &&
+            ((is_subword_type(Matcher::vector_element_basic_type(n->in(1))) &&
+              VM_Version::supports_avx512_bitalg()) ||
+             (is_non_subword_integral_type(Matcher::vector_element_basic_type(n->in(1))) &&
+              VM_Version::supports_avx512_vpopcntdq())));
```

Comment:
Please replace by:
predicate(is_pop_count_instr_target(Matcher::vector_element_basic_type(n->in(1))));
Also no need to check for UsePopCountInstruction any where as it is only meant for scalar popcount.